grammars/README.md

```diff
@@ -89,3 +89,13 @@ This guide provides a brief overview. Check out the GBNF files in this directory
 ```
 ./main -m <model> --grammar-file grammars/some-grammar.gbnf -p 'Some prompt'
 ```
+
+## Troubleshooting
```

Comment:
I like this section. After this gets merged in, I'll write a section on the dangers of left-recursion.

Reply:
We should probably also document the json->grammar converters here, I'll send that separately